bitwarden_license/bit-web/src/app/dirt/organization-integrations/integrations.component.ts

```diff
@@ -270,10 +273,58 @@ export class AdminConsoleIntegrationsComponent implements OnInit, OnDestroy {
         type: IntegrationType.EVENT,
         description: "crowdstrikeEventIntegrationDesc",
         canSetupConnection: true,
+        integrationType: 5, // Assuming 5 corresponds to CrowdStrike in OrganizationIntegrationType
       };
 
       this.integrationsList.push(crowdstrikeIntegration);
+
+      const datadogIntegration: Integration = {
+        name: OrganizationIntegrationServiceType.Datadog,
+        // TODO: Update link when help article is published
+        linkURL: "",
+        image: "../../../../../../../images/integrations/logo-datadog-color.svg",
+        type: IntegrationType.EVENT,
+        description: "datadogEventIntegrationDesc",
+        canSetupConnection: true,
+        integrationType: 6, // Assuming 6 corresponds to Datadog in OrganizationIntegrationType
+      };
+
+      this.integrationsList.push(datadogIntegration);
     }
+
+    // For all existing event based configurations loop through and assign the
+    // organizationIntegration for the correct services.
+    this.hecOrganizationIntegrationService.integrations$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((integrations) => {
+        // reset all integrations to null first - in case one was deleted
+        this.integrationsList.forEach((i) => {
+          i.organizationIntegration = null;
+        });
```

Comment:
Each service should reset its own integration type to prevent race conditions between services and/or loss of integration states.
```suggestion
        this.integrationsList.forEach((i) => {
          if (i.integrationType === OrganizationIntegrationType.Hec) {
            i.organizationIntegration = null;
          }
        });
```

To follow this pattern, line 240 should look like this:
```
        this.integrationsList.forEach((i) => {
          if (i.integrationType === OrganizationIntegrationType.Hec) {
            i.organizationIntegration = null;
          }
        });
```

and line 322 should be changed to this:

```
        this.integrationsList.forEach((i) => {
          if (i.integrationType === OrganizationIntegrationType.Datadog) {
            i.organizationIntegration = null;
          }
        });
```

and import this at the top:
`import { OrganizationIntegrationType } from "@bitwarden/bit-common/dirt/organization-integrations/models/organization-integration-type";`